vision/classification_and_detection/python/openimages.py

```diff
@@ -27,8 +27,9 @@ def __init__(self, data_path, image_list, name, use_cache=0, image_size=None,
         self.data_path = data_path
         self.pre_process = pre_process
         self.use_label_map=use_label_map
+
         if not cache_dir:
-            cache_dir = os.getcwd()
+            cache_dir = os.path.join(os.getcwd(), "validation", "data")
```

Comment:
This change seems to break the reference notebook
```
python3 python/main.py --profile retinanet-onnxruntime --mlperf_conf ../../mlperf.conf --model "/content/inference/vision/classification_and_detection/models/resnext50_32x4d_fpn.onnx" --dataset-path /content/inference/vision/classification_and_detection/open-images-v6-mlperf --output "/content/inference/vision/classification_and_detection/output/onnxruntime-gpu/retinanet" --time 1200 --max-latency 1.1 --accuracy
INFO:main:Namespace(accuracy=True, audit_conf='audit.config', backend='onnxruntime', cache=0, cache_dir=None, count=None, data_format=None, dataset='openimages-800-retinanet-onnx', dataset_list=None, dataset_path='/content/inference/vision/classification_and_detection/open-images-v6-mlperf', debug=False, find_peak_performance=False, inputs=['images'], max_batchsize=1, max_latency=1.1, mlperf_conf='../../mlperf.conf', model='/content/inference/vision/classification_and_detection/models/resnext50_32x4d_fpn.onnx', model_name='retinanet', output='/content/inference/vision/classification_and_detection/output/onnxruntime-gpu/retinanet', outputs=['boxes', 'labels', 'scores'], performance_sample_count=None, preprocessed_dir=None, profile='retinanet-onnxruntime', qps=None, samples_per_query=8, scenario='SingleStream', threads=2, time=1200, use_preprocessed_dataset=False, user_conf='user.conf')
ERROR:coco:no images in image list found
Traceback (most recent call last):
  File "python/main.py", line 615, in <module>
    main()
  File "python/main.py", line 482, in main
    ds = wanted_dataset(data_path=args.dataset_path,
  File "/content/inference/vision/classification_and_detection/python/openimages.py", line 113, in __init__
    raise ValueError("no images in image list found")
ValueError: no images in image list found
```

Reply:
Thank you @pgmpablo157321 for the review. The above change is now made conditional. The motivation of the change is to ensure preprocessed images going to the exact preprocessed_folder and same for dataset without any "prefix" so that external scripts can make use of the preprocess functions using a consistent API. Can you please link me to the reference notebook so that I can test it?

Reply:
You can find the reference notebook [here](https://github.com/mlcommons/inference/blob/master/vision/classification_and_detection/GettingStarted.ipynb). However, my comment was quite misleading. I just meant the change break the reference implementation if you follow the steps in the [readme](https://github.com/mlcommons/inference/tree/master/vision/classification_and_detection#:~:text=last%20month-,README.md,-Updating%20the%20readme). Sorry about that.

When I run: `./run_local.sh onnxruntime retinanet gpu --accuracy` I get the previous error.

Reply:
Replacing the line:
https://github.com/octoml/inference/blob/abc2e1a8fb857a9ce16d06fc1c2b8c95d673f88c/vision/classification_and_detection/python/openimages.py#L88
for:
`src = os.path.join(data_path, "validation", "data", image_name)`
Solves the issue, it seems that if you follow the steps of the readme, it is looking for the images in the root folder of the dataset

Reply:
Thank you @pgmpablo157321 for confirming. Actually the README is also not specific to retinanet. I believe the change is that DATA_DIR should be pointing to the directory where images are stored and not the parent folder where "validation/data" subfolder resides.